### PR TITLE
Right-justify rule codes and use code blocks around them

### PR DESCRIPTION
**Pull request by MicaelJarniac**
Right-justify rule codes on the Table of Contents (ToC), and wrap them in code blocks.

This should make it a lot easier and faster to visually scan through the many rule codes and find the one you're looking for.

I often found myself looking for a specific rule code, not knowing its name, and having to move my eyes something like this:

This made it very difficult to scan the rules, and very easy to accidentally miss the one I was looking for.

This does not break the anchors (`#pycodestyle-e-w`, for example) for the different rule sections.

This seems to work well on mobile.

## Before and After
 

Long rule names now look like this: 
image

And section names look like this:
image

### CODE REVIEW

**Comment by MicaelJarniac** [comment]
Minor issue with one of the changes, the browser tab title shows the backticks:
image

File: crates/ruff_dev/src/generate_docs.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ pub(crate) fn main(args: &Args) -> Result<()> {
     for rule in Rule::iter() {
         if let Some(explanation) = rule.explanation() {
             let mut output = String::new();
-            output.push_str(&format!("# {} ({})", rule.as_ref(), rule.noqa_code()));
+            output.push_str(&format!("# (`{}`) {}", rule.noqa_code(), rule.as_ref()));
```
Comment (MicaelJarniac) [inline crates/ruff_dev/src/generate_docs.rs:28]:
```suggestion
            output.push_str(&format!("# ({}) {}", rule.noqa_code(), rule.as_ref()));
```
To fix the browser tab title showing the backticks:
image
I'm not sure which would be preferable, having the code block or fixing the tab title.

**Comment by MichaReiser** [comment]
Would it be possible to right algin the rule codes instead? Because we now have the opposite issue: The rule names are no longer aligned.

**Comment by MicaelJarniac** [comment]
> Would it be possible to right algin the rule codes instead? Because we now have the opposite issue: The rule names are no longer aligned.

As far as I can tell, the sidebar is automatically generated based on the header tags of the page, that are also automatically generated, and they're Markdown.

I'll see if I can figure out something, but I suspect it won't be easy to do.

**Comment by MicaelJarniac** [comment]
@MichaReiser thanks for your suggestion! It took quite a while, but I managed to do it, and I agree it's much better.

**Comment by MicaelJarniac** [comment]
Docs generation isn't working on the GitHub Action, for whatever reason. I'm trying to figure it out...

**Comment by MicaelJarniac** [comment]
I haven't managed to figure it out yet.
[Error](https://github.com/astral-sh/ruff/actions/runs/5447999581/jobs/9910664860?pr=5450#step:10:14)
```
ERROR    -  Config value 'markdown_extensions': Failed to load extension 'custom-toc'.
ModuleNotFoundError: No module named 'custom-toc'
Aborted with 1 Configuration Errors!
Error: Process completed with exit code 1.
```

**Comment by MicaelJarniac** [comment]
Okay, so it seems like setting a Python version for the docs workflow explicitly fixed it.
I've set it to `3.11`, but I suspect earlier versions _might_ work.
If `3.11` is too bleeding-edge for building the docs, we could try another version.

**Comment by charliermarsh** [comment]
I'm getting:

```
ERROR    -  Config value 'markdown_extensions': Failed to load extension
            'custom-toc'.
            ModuleNotFoundError: No module named 'custom-toc
```

When running locally. I ran `pip install -r docs/requirements.txt` on `Python 3.11.2`. Any ideas?

**Comment by MicaelJarniac** [comment]
This is the error that was happening on the GitHub Action for `mkdocs`, and I kinda "fixed" it by forcing the workflow to Python 3.11.

Locally I can install it with Python 3.11.3 and it works fine.

Could you try installing the subpackage directly, with `pip install docs/custom-extension`, and share the output?

I'm not entirely sure what's the problem, or how to fix it.

One thing I considered was to actually publish this custom extension, but I'm not sure if it'd be desirable, as it'd then become another dependency.

I'll try installing Python 3.11.2 here and see if I can reproduce it.

**Comment by MicaelJarniac** [comment]
Another thing worth testing, `pip install -U pip setuptools wheel`.

**Comment by MicaelJarniac** [comment]
Tested with Python 3.11.2 and it still works here. I have no idea what's going on.

**Comment by charliermarsh** [comment]
All good, I will take another look.

**Comment by MicaelJarniac** [comment]
I tried merging `main` back here and now it's giving me SSH key errors of some sort.
`Error: The ssh-private-key argument is empty. Maybe the secret has not been configured, or you are using a wrong secret name in your workflow file.`

**Comment by MichaReiser** [comment]
I like the new layout. @charliermarsh how do you feel about copying over the extension and modifying it in place?

**Comment by charliermarsh** [comment]
I appreciate this work, but my current feeling is that it's not worth forking + maintaining a fork for this level of improvement. I marginally prefer the new layout, but to put it differently, I'm not sure that the complexity outweighs the benefit. If we could upstream the change somehow, I'd be open to reconsidering. But just as an example, as-is, this doesn't work with the `typeset` plugin, which is sadly on Insiders (I know, I know) but will be mainstreamed in a future release, and which is useful to enable for (e.g.) the settings page:

Screen Shot 2023-07-06 at 5 15 16 PM

So we'd then need to get rid of the `typeset` plugin to support this. Overall, it feels heavyweight to maintain this fork, custom package, etc., and we'll have to think about compatibility going forward. I'm sorry for this outcome, I hate saying no to PRs, but it feels like the responsible decision here from a maintainability perspective.

**Comment by MicaelJarniac** [comment]
@charliermarsh what about the first version of this PR, that simply puts the rule code before the rule name?

image

**Comment by MicaelJarniac** [comment]
Also, I'd like to look into upstreaming the patched `toc` plugin. Is there a way I can experiment with this incompatible `typeset` plugin you mentioned, so I can try to make it compatible?

**Pull request closed without merging**